src/install/install.zig

```diff
@@ -10089,162 +10104,466 @@ pub const PackageManager = struct {
-            .result => |fd| fd,
-            .err => |e| {
-                Output.prettyErrorln(
-                    "\n<r><red>error<r>: {}\n",
-                    .{
-                        e.toSystemError(),
-                    },
-                );
-                Output.flush();
-                return;
-            },
+                        const entrypath = bun.strings.fromWPath(pathbuf[0..], entry.path);
+                        pathbuf[entrypath.len] = 0;
+                        const entrypathZ = pathbuf[0..entrypath.len :0];
+
+                        if (bun.sys.renameatConcurrently(
+                            bun.toFD(destination_dir_.fd),
+                            entrypathZ,
+                            bun.toFD(destination_dir_.fd),
+                            tmpname,
+                        ).asErr()) |e| {
+                            Output.prettyError("<r><red>error<r>: copying file {}", .{e});
+                            Global.crash();
+                        }
+
+                        var in_file = openFile(entry.dir, basename, .{ .mode = .read_only }) catch |e| {
+                            Output.prettyError("<r><red>error<r>: opening node_modules file {s} {s} ()", .{ @errorName(e), entrypath });
+                            Global.crash();
+                        };
+                        var in_file_close = true;
+                        defer if (in_file_close) in_file.close();
+
+                        const mode = in_file.mode() catch |e| {
+                            Output.prettyError("<r><red>error<r>: failed to get file mode {s} ", .{@errorName(e)});
+                            Global.crash();
+                        };
+                        var outfile = createFile(destination_dir_, entrypath, .{ .mode = mode }) catch |e| {
+                            Output.prettyError("<r><red>error<r>: failed to create file {s} ({s})", .{ entrypath, @errorName(e) });
+                            Global.crash();
+                        };
+                        var out_file_close = true;
+                        defer if (out_file_close) outfile.close();
+
+                        const infile_path = bun.path.joinStringBufWZ(buf1, &[_][]const u16{ in_dir, entry.path }, .auto);
+                        const outfile_path = bun.path.joinStringBufWZ(buf2, &[_][]const u16{ out_dir, entry.path }, .auto);
+
+                        in_file_close = false;
+                        out_file_close = false;
+                        in_file.close();
+                        outfile.close();
+                        bun.copyFileWithState(infile_path, outfile_path, &copy_file_state) catch |err| {
+                            Output.prettyError("<r><red>{s}<r>: copying file {}", .{ @errorName(err), bun.fmt.fmtOSPath(entry.path, .{}) });
+                            Global.crash();
+                        };
+
+                        continue;
+                    }
```

Comment:
Can you move the rest of this into an `else if (comptime Environment.isPosix)`? It's harder to tell where windows ends and where posix ends